File: ProAgil.Domain/DataModels/Event.cs

using System;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;

namespace ProAgil.Domain.DataModels
{
    public class Event
    {
        [Key]
        public int Id { get; set; }
        public String Local { get; set; }
        public DateTime EventDate { get; set; }
        public String Subject { get; set; }
        public int GuestsQty { get; set; }
        public String Phone { get; set; }
        public String Email { get; set; }
        public String ImageURL { get; set; }


        #region EF Entity Relations
        public List<EventLot> Lots { get; set; }
        public List<SocialMedia> SocialMedias { get; set; }
        public List<SpeakerEvent> SpeakersEvents { get; set; }

        #endregion
        public Event() { }


    }
}

File: ProAgilAPI-Ajs/src/app/models/Event.ts
import { EventLot } from './EventLot';
import { SocialMedia } from './SocialMedia';
import { Speaker } from './Speaker';

export interface Event {
    // use lower case on variable name because the angular return is on lCase. 
    id: number; // equals int in C#
    local: string;
    eventDate: Date; // equals DateTime in C#
    subject: string;
    guestsQty: number;
    phone: string;
    email: string;
    imageURL: string;
    lots: EventLot[];
    socialMedias: SocialMedia[];
    speakersEvents: Speaker[];

}
